src/Bridge/Doctrine/DBAL/Types/AbstractEnumType.php

```diff
@@ -14,12 +14,16 @@
 use Doctrine\DBAL\Types\Type;
 use Elao\Enum\EnumInterface;
 
+/**
+ * @template T of EnumInterface
+ */
 abstract class AbstractEnumType extends Type
 {
     /**
      * The enum FQCN for which we should make the DBAL conversion.
      *
-     * @return class-string<EnumInterface>
+     * @return string&EnumInterface
```

Comment:
this line seems wrong to me

Reply:
It's wrong indeed, but it's on purpose for autocompletion to work here: https://github.com/Elao/PhpEnums/blob/c533ec1463fd8aa42b7afd771a9c7cd7070141b4/src/Bridge/Doctrine/DBAL/Types/AbstractEnumType.php#L72

but not sure if we should keep it here on the method signature or locally on usage (or remove completly)

Reply:
is `AbstractEnumSQLDeclarationType` missing an annotation?

Reply:
for example, how should i annotate this class?

```php
final class FormUsageEnumType extends AbstractEnumSQLDeclarationType
{
    public const NAME = 'form_usage';

    protected function getEnumClass(): string
    {
        return FormUsage::class;
    }

    public function getName(): string
    {
        return self::NAME;
    }
}
```

phpstan will complain:

```Method FormUsageEnumType::getEnumClass() return type with generic interface Elao\Enum\EnumInterface does not specify its types: T
```

Reply:
should `AbstractEnumSQLDeclarationType` have 
```php

/**
 * @template T of EnumInterface
 * @extends AbstractEnumType<T>
 */
abstract class AbstractEnumSQLDeclarationType extends AbstractEnumType {}
```

?

Reply:
I am doing something like this 


```
/**
 * @extends AbstractEnumType<Status>
 */
final class StatusType extends AbstractEnumType
{
    private const NAME = 'status';

    /** {@inheritdoc} */
    protected function getEnumClass(): string
    {
        return Status::class;
    }

    /** {@inheritdoc} */
    public function getName(): string
    {
        return static::NAME;
    }

    /** {@inheritdoc} */
    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform): string
    {
        return 'TEXT';
    }
}
```